radicle/src/cob/op.rs

```diff
@@ -34,6 +37,42 @@ impl OpId {
     }
 }
 
+impl fmt::Display for OpId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}/{}", self.1, self.0)
+    }
+}
+
+/// Error decoding an operation from an entry.
+#[derive(Error, Debug)]
+pub enum OpIdError {
+    #[error("invalid Op ID")]
```

Comment:
It's quite annoying when errors just say that it's invalid. Can we add the different variants of why it's gone wrong? i.e. missing '/', incorrect Lamport clock, and incorrect ActorId.

Reply:
The difference here is are the errors for developers or users?

If a user inputs a bad Op ID its because they copy+pasted it wrong.  The error would be redundant at this point.

Reply:
I would say both and I'd argue that a lot of users are "developers" in the general sense so it will help them also diagnose when things go wrong.

Say they paste something they thought was an Op ID, which do you think is more useful:
1) "invalid Op ID"
2) "expected an Op ID of the form '<actor id>/<clock>', but the '<actor id>' was malformed"

I know which one I would like to see :)

Reply:
I would prefer to see the first in logging, the later comes across as verbose.  I only see information on how the formatting should be being useful if they are handcrafted, or if they are empty.

Reply:
At the very least I would report what the string was then. In my mind, I'm comparing this to getting back `File error` and not knowing which file name had an error.

Reply:
I've updated the error to distinguish between an empty string and a badly formatted Op ID string

Reply:
You can likely google "useful error messages" and I'd say you'd find similar advice across any articles. One I read a while back (would need to revisit) is [this post by Tikhon Jelvis](https://jelv.is/blog/Structure-your-Errors/). I'm of the opinion that more information is better when it comes to errors -- you can filter information presented to you but it's harder to _add_ information.

Feel free to proceed anyway. I'll take another look at the PR.

Reply:
Thanks, Ill check it out.  I find recommendations better because the signal to noise ratio is too high.